packages/shared/src/services/performance-metrics-methods.ts

```diff
@@ -0,0 +1,52 @@
+import { Analytics } from '@deriv-com/analytics';
+
+declare global {
+    interface Window {
+        performance_metrics: {
+            create_ctrader_account_time: number;
+            create_dxtrade_account_time: number;
+            create_mt5_account_time: number;
+            load_cashier_time: number;
+            login_time: number;
+            redirect_from_deriv_com_time: number;
+            signup_time: number;
+            switch_currency_accounts_time: number;
+            switch_from_demo_to_real_time: number;
+            switch_from_real_to_demo_time: number;
+        };
+    }
+}
+
+// action type will be updated based on the type from Analytics package when it will be updated
+export const startPerformanceEventTimer = (action: keyof typeof global.Window.prototype.performance_metrics) => {
+    if (!window.performance_metrics) {
+        window.performance_metrics = {
+            create_ctrader_account_time: 0,
+            create_dxtrade_account_time: 0,
+            create_mt5_account_time: 0,
+            load_cashier_time: 0,
+            login_time: 0,
+            redirect_from_deriv_com_time: 0,
+            signup_time: 0,
+            switch_currency_accounts_time: 0,
+            switch_from_demo_to_real_time: 0,
+            switch_from_real_to_demo_time: 0,
+        };
+    }
+
+    window.performance_metrics[action] = Date.now();
+};
+
+export const setPerformanceValue = (action: keyof typeof global.Window.prototype.performance_metrics) => {
+    if (window.performance_metrics?.[action]) {
+        const value = (Date.now() - window.performance_metrics[action]) / 1000;
+        window.performance_metrics[action] = 0;
+
+        const event_name = 'ce_performance_metrics';
```

Comment:
can we have a more elaborate naming for this?
`ce_performance_metrics`

Reply:
@habib-deriv we can set any name what we want :) just suggest and I'll change it if you want. But all Events in this [library](https://github.com/binary-com/deriv-analytics/blob/master/src/types.ts) starts from `ce`

Reply:
nope @habib-deriv it's requirement for analytics usage. trackingEvent pushing this data to bigtable database